File: HLC01_03Java/src/main/java/org/example/reservaHotel02/Room.java
package org.example.reservaHotel02;

public class Room {
    private final String number;
    private final double price;
    private boolean booked;


    public Room(String number, double price) {
        this.number = number;
        this.price = price;
        this.booked = false;
    }


    public String getNumber() {
        return number;
    }


    public double getPrice() {
        return price;
    }


    public boolean isBooked() {
        return booked;
    }


    public void book() {
        this.booked = true;
    }


    public void unBook() {
        this.booked = false;
    }


    @Override
    public String toString() {
        return "Habitación: " + number + " - Precio: " + price + " - Estado: " + (booked ? "Reservada" : "Libre");
    }
}

File: HLC01_03Kotlin/src/main/kotlin/reservaHotel02/Room.kt
package org.example.reservaHotel02

data class Room(
    val number: String,
    val price: Double,
) {
    var booked: Boolean = false
        private set


    fun book() {
        booked = true
    }


    fun unBook() {
        booked = false
    }


    override fun toString(): String {
        return "Habitación: $number - Precio: $price - Estado: ${if (booked) {"Reservada"} else {"Libre"}}"
    }
}